presto-common/src/test/java/com/facebook/presto/common/TestPrestoVersion.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common;
+
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestPrestoVersion
+{
+    private static final PrestoVersion TEST_PRESTO_VERSION_1 = new PrestoVersion("0.274-20220517.171619-22");
+    private static final PrestoVersion TEST_PRESTO_VERSION_2 = new PrestoVersion("0.274-SNAPSHOT-5d0ba93");
+    private static final PrestoVersion TEST_PRESTO_VERSION_3 = new PrestoVersion("0.274");
+    private static final PrestoVersion TEST_PRESTO_VERSION_4 = new PrestoVersion("0.274.1");
+    private static final PrestoVersion TEST_PRESTO_VERSION_5 = new PrestoVersion("0.273");
+    private static final PrestoVersion TEST_PRESTO_VERSION_6 = new PrestoVersion("0.274.1");
+    private static final PrestoVersion TEST_PRESTO_VERSION_7 = new PrestoVersion("0.274-20220518.025123-29");
+
+    @Test
+    public void testPrestoEquals()
+    {
+        assertEquals(TEST_PRESTO_VERSION_4.hashCode(), TEST_PRESTO_VERSION_6.hashCode());
```

Comment:
since hashCode is straightforward you can remove this. If you want to keep it move it to a different method since this one is for testing equals.